src/Caching/Storages/PeclMemcachedStorage.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+/**
+ * This file is part of the Nette Framework (https://nette.org)
+ * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
+ */
+
+namespace Nette\Caching\Storages;
+
+use Nette;
+use Nette\Caching\Cache;
+
+
+/**
+ * Memcached storage using memcached extension.
+ */
+class PeclMemcachedStorage extends Nette\Object implements Nette\Caching\IStorage
+{
+	/** @internal cache structure */
+	const META_CALLBACKS = 'callbacks',
+		META_DATA = 'data',
+		META_DELTA = 'delta';
+
+	/** @var \Memcached */
+	private $memcache;
+
+	/** @var string */
+	private $prefix;
+
+	/** @var IJournal */
+	private $journal;
+
+
+	/**
+	 * Checks if Memcached extension is available.
+	 * @return bool
+	 */
+	public static function isAvailable()
+	{
+		return extension_loaded('memcached');
+	}
+
+
+	public function __construct($host = 'localhost', $port = 11211, $prefix = '', IJournal $journal = NULL)
+	{
+		if (!static::isAvailable()) {
+			throw new Nette\NotSupportedException("PHP extension 'memcache' is not loaded.");
+		}
+
+		$this->prefix = $prefix;
+		$this->journal = $journal;
+		$this->memcache = new \Memcached;
+		if ($host) {
+			$this->addServer($host, $port);
+		}
+	}
+
+
+	public function addServer($host = 'localhost', $port = 11211, $timeout = 1)
+	{
+		if ($this->memcache->addServer($host, $port, 1) === FALSE) {
+			$error = error_get_last();
+			throw new Nette\InvalidStateException("Memcached::addServer(): $error[message].");
```

Comment:
Well, 
- app without cache actually can kill the server, not only one request
- this is copy & past & modify from the current MemcachedStorage.